manifests/v2/base/webhook/manifests.yaml

```
@@ -0,0 +1,129 @@
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: mutating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /mutate-kubeflow-org-v2alpha1-clustertrainingruntime
+  failurePolicy: Fail
+  name: mclustertrainingruntime.training-operator.kubeflow.org
+  rules:
+  - apiGroups:
+    - kubeflow.org
+    apiVersions:
+    - v2alpha1
+    operations:
+    - CREATE
+    resources:
+    - clustertrainingruntimes
+  sideEffects: None
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /mutate-kubeflow-org-v2alpha1-trainingruntime
+  failurePolicy: Fail
+  name: mtrainingruntime.training-operator.kubeflow.org
+  rules:
+  - apiGroups:
+    - kubeflow.org
+    apiVersions:
+    - v2alpha1
+    operations:
+    - CREATE
+    resources:
+    - trainingruntimes
+  sideEffects: None
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /mutate-kubeflow-org-v2alpha1-trainjob
+  failurePolicy: Fail
+  name: mtrainjob.training-operator.kubeflow.org
+  rules:
+  - apiGroups:
+    - kubeflow.org
+    apiVersions:
+    - v2alpha1
+    operations:
+    - CREATE
+    resources:
+    - trainjobs
+  sideEffects: None
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: validating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /validate-kubeflow-org-v2alpha1-clustertrainingruntime
+  failurePolicy: Fail
+  name: vclustertrainingruntime.training-operator.kubeflow.org
```

Should we simplify it ?

```suggestion
  name: clustertrainingruntime.training.kubeflow.org
```

---

To avoid the conflict between the defaulter and the validator, let's use the "validator" prefix.
Additionally, could we use the "training-operaotr.kubeflow.org" domain since we have 2 components, training-operator and mpi-operator for the training?

---

Done.